libp2p/stream/connection.nim

```diff
@@ -33,6 +33,7 @@ type
     peerInfo*: PeerInfo
     observedAddr*: Multiaddress
     upgraded*: Future[void]
+    outbound*: bool
```

Comment:
why do we have both `dir` as a direction enum and outbound as a `bool`?

Reply:
well as you can see from the diff I used to use `dir`, but that's not the actual wire direction.

Reply:
well, the point was more, why is it not using the direction enum?

also, I presume that what we want to capture here is whether or not the underlying connection is outbound or not, and not the mplex stream on top - this should be documented clearly along with a reason why / pointer to the spec, because it's hard to follow. 

finally, the name could be used to indicate some of this as well, ie `transportDirection` or something - conceptually, it goes a bit against the abstraction of channels and connections being layered - basically, once a transport is established, one doesn't "normally" care in what direction the underlying stream was established, or indeed how many layers of envelopes there are (in theory, on could run mplex inside mplex inside noise)